### DISCUSSION TITLE
Insert if it doesn't exist, otherwise update

### DISCUSSION THREAD

**Discussion by eliluong** [msiemens/tinydb#442] [Q&A] [answered]
Hi, if I have database structured in the form of `{'key': key,}`, and if `key` already exists, then I want to update to add `'other': 'other_data'` but if `key` does not exist, then I want to do an insert.

Is there a way to do this in TinyDB? I notice if I insert `{'key': key,'other': 'other_data'}` it will make another entry, because it is using its own unique numerical key. My solution is to always do a search for `key` and if it returns nothing then do an insert, otherwise do an update.

Thanks.

**Comment by msiemens** [comment, marked as answer]
@eliluong This sounds like the `upsert` operator: https://tinydb.readthedocs.io/en/latest/usage.html#upserting-data

Basically, you pass a query and a document. If the query matches any document that's already in the database, it will update it using the document you gave it. If there is _no_ result, it will insert the provided document as a new one.

Does that work for you?

**Reply by eliluong** [reply]
Thanks, @msiemens I think this is exactly what I am looking for. Did not notice this when I read over the documentation. Now I know about `upsert` in databases :)
